src/lib.rs

```diff
@@ -38,13 +38,37 @@ struct Package {
 }
 
 /// Get the repository name from `Cargo.toml`.
+/// Fallbacks to directory name if `Cargo.toml` does not exists.
 pub fn read_repo(dir: &str) -> ::Result<String> {
   let mut dir = PathBuf::from(dir);
   dir.push("Cargo.toml");
-  let cargo_toml = fs::read_to_string(dir).context(::ErrorKind::Other)?;
-  let config: Config =
-    toml::from_str(&cargo_toml).context(::ErrorKind::Other)?;
-  Ok(config.package.repository)
+
+  if dir.as_path().exists() {
+    let cargo_toml = fs::read_to_string(dir).context(::ErrorKind::Other)?;
+    let config: Config =
+      toml::from_str(&cargo_toml).context(::ErrorKind::Other)?;
+    Ok(config.package.repository)
+  } else {
+    dir.pop();
+    match dir.file_name() {
+      Some(file_name) => {
+        let config: Config = Config {
```

Comment:
There is no need to add the `:Config` section here. The type signature is inferred automatically. Instead do:
```rust
let config = Config {
```